artifactory/commands/buildinfo/addgit_test.go

```diff
@@ -62,32 +64,75 @@ func TestBuildAddGitSubmodules(t *testing.T) {
 		t.Run(test, func(t *testing.T) {
 			tmpBuildName := test + "-Build-" + strconv.FormatInt(time.Now().Unix(), 10)
 			err := runBuildAddGit(t, tmpBuildName, "1", projectPath, test == "dotGitProvided")
+			require.NoError(t, err)
+			partials := getBuildInfoPartials(t, tmpBuildName, "1", "")
+			assertVcsSubmodules(t, partials)
+		})
+	}
+}
+
+func TestBuildAddGitVCSDetails(t *testing.T) {
+	bagTests := []struct {
+		name        string
+		originalDir string
+		revision    string
+		branch      string
+		message     string
+	}{
+		{"Test vcs details without branch", withGit, "6198a6294722fdc75a570aac505784d2ec0d1818", "", "TEST-2 - Adding text to file1.txt"},
+		{"Test vcs details with branch", withBranch, "b033a0e508bdb52eee25654c9e12db33ff01b8ff", "master", "TEST-4 - Adding text to file2.txt"}}
+
+	for _, test := range bagTests {
+		t.Run(test.name, func(t *testing.T) {
+			baseDir, dotGitPath := tests.PrepareDotGitDir(t, test.originalDir, filepath.Join("..", "testdata"))
+			buildDir := getBuildDir(t)
+			checkFailureAndClean(t, buildDir, dotGitPath, test.originalDir)
+			err := runBuildAddGit(t, buildName, "1", baseDir, true)
 			if err != nil {
 				return
 			}
-			partials := getBuildInfoPartials(t, tmpBuildName, "1", "")
-			assertVcsSubmodules(t, partials)
+			partials := getBuildInfoPartials(t, buildName, "1", "")
+			checkVCSDetails(partials, test.revision, test.branch, test.message, t)
+			checkFailureAndClean(t, buildDir, dotGitPath, test.originalDir)
+			tests.RemovePath(buildDir, t)
+			tests.RenamePath(dotGitPath, filepath.Join(filepath.Join("..", "testdata"), test.originalDir), t)
 		})
 	}
 }
 
+func checkVCSDetails(partials buildinfo.Partials, revision, branch, message string, t *testing.T) {
+	for _, partial := range partials {
+		if partial.VcsList != nil {
+			for _, vcs := range partial.VcsList {
+				assert.Equal(t, revision, vcs.Revision)
+				assert.Equal(t, branch, vcs.Branch)
+				assert.Equal(t, message, vcs.Message)
+			}
+		} else {
+			t.Error("VCS cannot be nil")
+			break
+		}
+	}
+}
+
 func assertVcsSubmodules(t *testing.T, partials buildinfo.Partials) {
 	assert.Len(t, partials, 1)
 	vcsList := partials[0].VcsList
 	assert.NotNil(t, vcsList)
 	assert.Len(t, vcsList, 1)
 	curVcs := vcsList[0]
 	assert.Equal(t, "https://github.com/jfrog/jfrog-cli.git", curVcs.Url)
-	assert.Equal(t, "d63c5957ad6819f4c02a817abe757f210d35ff92", curVcs.Revision)
+	assert.Equal(t, "6198a6294722fdc75a570aac505784d2ec0d1818", curVcs.Revision)
 	assert.Equal(t, "submodule", curVcs.Branch)
+	assert.Equal(t, "TEST-2 - Adding text to file1.txt", curVcs.Message)
 }
 
 // Clean the environment if fails
-func checkFailureAndClean(t *testing.T, buildDir string, oldPath string) {
+func checkFailureAndClean(t *testing.T, buildDir, oldPath, originalDir string) {
```

Comment:
Let's use a cleanUp function instead of this function:
```go
func cleanUp(t *testing.T, buildDir, dotGitPath, originalDir string) {
	if buildDir != "" {
		tests.RemovePath(buildDir, t)
	}
	if dotGitPath != "" {
		tests.RenamePath(dotGitPath, filepath.Join(filepath.Join("..", "testdata"), originalDir), t)
	}
}
```
Call the cleanUp from `TestBuildAddGitVCSDetails` and `runTest` using "defer", early as possible. For example:
```go
baseDir, dotGitPath := tests.PrepareDotGitDir(t, test.originalDir, filepath.Join("..", "testdata"))
buildDir := getBuildDir(t)
defer cleanUp(t, buildDir, dotGitPath, test.originalDir)
```